appmap/__init__.py

```diff
@@ -1,38 +1,43 @@
 """AppMap recorder for Python"""
-
-from _appmap import generation  # noqa: F401
-from _appmap.env import Env  # noqa: F401
-from _appmap.importer import instrument_module  # noqa: F401
-from _appmap.labels import labels  # noqa: F401
-from _appmap.noappmap import decorator as noappmap
-from _appmap.recording import Recording  # noqa: F401
-
-try:
-    from . import django  # noqa: F401
-except ImportError:
-    pass
-
-try:
-    from . import flask  # noqa: F401
-except ImportError:
-    pass
-
-try:
-    from . import fastapi  # noqa: F401
-except ImportError:
-    pass
-
-try:
-    from . import uvicorn  # noqa: F401
-except ImportError:
-    pass
-
-# Note: pytest integration is configured as a pytest plugin, so it doesn't need to be imported here
-
-# unittest is part of the standard library, so it should always be importable (and therefore doesn't
-# need to be in a try .. except block)
-from . import unittest  # noqa: F401
-
-
-def enabled():
-    return Env.current.enabled
+import os
+
+# Note that we need to guard these imports with a conditional, rather than
+# putting them in a function and conditionally calling the function. If we
+# execute the imports in a function, the modules all get put into the funtion's
+# globals, rather than into appmap's globals.
+if os.environ.get("APPMAP", "false").upper() == "TRUE":
```

Comment:
Due to the drastic nature of this change, this would be a major version revision. 

The library on the current major version should warn loudly that this change is going to happen. 

We will also need to notify Python users as robustly as we can, eg with an email.

Reply:
> Due to the drastic nature of this change, this would be a major version revision.
Ok. I'll mark this a breaking change.

> 
> The library on the current major version should warn loudly that this change is going to happen.
I'll open separate PR for this that we'll merge before this one.

> 
> We will also need to notify Python users as robustly as we can, eg with an email.
I'm not sure what the mechanism for sending such an email would be, so I'll need some help with this.

Reply:
I've added a warning that gets displayed when the user is relying on the default behavior.

For example, when running `pytest`, they'll see this:
```
======================== 1 xfailed, 1 warning in 1.46s =========================
pytest-xprocess reminder::Be sure to terminate the started process by running 'pytest --xkill' if you have not explicitly done so in your fixture with 'xprocess.getinfo(<process_name>).terminate()'.
/Users/ajp/src/applandinc/appmap-python/appmap/pytest.py:30: DeprecationWarning: 

The APPMAP environment variable is unset. Your code will be
instrumented and recorded according to the configuration in appmap.yml.

Starting with version 2, this behavior will change: when APPMAP is
unset, no code will be instrumented. You will need to use the
appmap-python script to run your application, or explicitly set
APPMAP.

Visit https://appmap.io/docs/reference/appmap-python.html#appmap-python-script for more
details.

  Env.current.warn_enabled_by_default()
[2024-04-23 09:06:30,521] WARNING _appmap.metadata: Failed running 'git rev-parse HEAD', fatal: not a git repository (or any of the parent directories): .git

PASSED
```
This is the standard way that`pytest` reports warnings that occur during tests.

When running a Django app, they'll see this:
```
/Users/ajp/src/applandinc/appmap-python/.direnv/python-3.10.13/bin/python manage.py runserver 127.0.0.1:8000
[2024-04-23 09:09:30,794] WARNING _appmap.env: Setting APPMAP_OUTPUT_DIR is not supported
[2024-04-23 09:09:30,854] WARNING _appmap.env: Setting APPMAP_OUTPUT_DIR is not supported
[2024-04-23 08:09:32,515] WARNING _appmap.env: Setting APPMAP_OUTPUT_DIR is not supported
[2024-04-23 08:09:32,551] WARNING _appmap.env: Setting APPMAP_OUTPUT_DIR is not supported
Performing system checks...

System check identified no issues (0 silenced).
/Users/ajp/src/applandinc/appmap-python/_appmap/web_framework.py:250: DeprecationWarning: 

The APPMAP environment variable is unset. Your code will be
instrumented and recorded according to the configuration in appmap.yml.

Starting with version 2, this behavior will change: when APPMAP is
unset, no code will be instrumented. You will need to use the
appmap-python script to run your application, or explicitly set
APPMAP.

Visit https://appmap.io/docs/reference/appmap-python.html#appmap-python-script for more
details.

  Env.current.warn_enabled_by_default()
April 23, 2024 - 08:09:33
Django version 5.0.4, using settings 'djangoapp.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
```
Again, standard reporting.

The user will also see a warning when doing process recording, as well as code-block recording.

You asked for the help text to show an example command line, but the way the Standard Library's `warnings` module works, as well as the way `argv` is handled, didn't seem to allow this. I'm hoping that linking to the doc is an acceptable compromise.